### ISSUE TITLE
kata-ctl: Re-enable network check on s390x

### ISSUE THREAD

**Issue by jodh-intel** [kata-containers/kata-containers#5438]
The kata-ctl network check needs to be disabled on s390x using #5435 as that platform apparently doesn't support rust-tls (yet).

If and when rust-tls is supported, the `reqwest` HTTP crate can be used unconditionally by `kata-ctl`.

/cc @BbolroC.

**Comment by BbolroC** [comment]
Thanks for looking into the issue. I will keep an eye on the support.

**Comment by hbrueckner** [comment]
fyi, the TLS library used by rust-tls, `ring` lacks support for s390x. There is s390x basic support PR pending since last year: https://github.com/briansmith/ring/pull/1297

Not sure if there are other TLS alternatives for doing the tests or if we need to work and wait on above dependency.

**Comment by hbrueckner** [comment]
Hi @jodh-intel 

I briefly looked at https://github.com/seanmonstar/reqwest and it seems that `reqwest` can use the native TLS crate (openssl) and being `rustls-tls` an optional feature. For s390x, I wonder whether simplying removing the `rustls-tls` feature would be the easiest option to re-enable the test case. Wdyt?

**Comment by jodh-intel** [comment]
@hbrueckner, @BbolroC - I had problems using reqwest with the native libs on x86_64 due to our rust toolchain using musl as the libc, hence switched to rust-tls.

I'm trying to land the admittedly rather ugly but I think working fix (#5439) to unbreak the Kata CI.

However, it would be great if one of you who has access to s380x h/w could raise a follow-up PR that fixed the problem "more properly". Maybe we could have something like this in the manifest:

```toml
[target.'cfg(not(target_arch = "s390x"))'.dependencies]
reqwest = { version = "0.11", default-features = false, features = ["json", "blocking", "rustls-tls"] }

[target.'cfg(target_arch = "s390x")'.dependencies]
reqwest = { version = "0.11", default-features = false, features = ["json", "blocking"] }
```

But I think you'd still need to find a solution to the musl problem. An alternative might be that we build `kata-ctl` using the standard libc though as I mentioned on https://github.com/jodh-intel/kata-containers/pull/10#pullrequestreview-1133119110.
